src/python/pants/rules/core/list_targets.py

```diff
@@ -82,7 +83,25 @@ def print_documented(target):
   yield List(exit_code=0)
 
 
+class Roots(LineOriented, Goal):
+  """List the repo's registered source roots."""
+  name = 'rootsv2'
+
+
+@console_rule(Roots, [Console, Roots.Options, SourceRootConfig])
+def list_roots(console, options, source_root_config):
+  all_roots = SourceRoots(source_root_config).all_roots()
+  print("ALL ROOTS: ", list(all_roots))
+  with Roots.line_oriented(options, console) as (print_stdout, print_stderr):
+    for src_root in all_roots:
+      all_langs = ','.join(sorted(src_root.langs))
+      print_stdout('{}: {}'.format(src_root.path, all_langs or '*'))
```

Comment:
(I know this is still a WIP). Can use an f-string here:
```suggestion
      print_stdout(f"{src_root.path}: {all_langs or '*'}")
```